Sources/Datadog/DatadogCore/DatadogCore.swift

```diff
@@ -6,33 +6,60 @@
 
 import Foundation
 
+/// Feature-agnostic SDK configuration.
+internal typealias CoreConfiguration = FeaturesConfiguration.Common
+
+/// Feature-agnostic set of dependencies powering Features storage, upload and event recording.
+internal typealias CoreDependencies = FeaturesCommonDependencies
+
+/// A shim interface for uniforming V1 Features and letting their generic initialization in `DatadogCore`.
+internal protocol V1Feature {
+    /// The configuration specific to this Feature.
+    /// In V2 this will likely become a part of the public interface for the Feature module.
+    associatedtype Configuration
+
+    init(
+        storage: FeatureStorage,
+        upload: FeatureUpload,
+        configuration: Configuration,
+        commonDependencies: FeaturesCommonDependencies
+    )
```

Comment:
I like the idea of type constraint on v1 Features, but I'm wondering if we could get rid of the `init` and `associatedtype` 🤔 We could simply use convention to initialise v1 features and be able to register `V1Feature` conforming instances:
```swift
public protocol DatadogCoreProtocol {
    // MARK: V1 interface

    func register<T>(feature instance: T?) where T: V1Feature

    func feature<T>(_ type: T.Type) -> T? where T: V1Feature
}
```
It would then be easier to add other properties to `V1Feature`, such as storage and upload. WDYT?

Reply:
Indeed 👍. I tackled this by renaming `V1Feature` to `V1FeatureInitializable` in https://github.com/DataDog/dd-sdk-ios/pull/871/commits/feb5766cd7554bacbb3e3ab747793b545319b96d. This is only conventional change, but will allow for eventually having non-constrained `V1Feature` protocol:

```swift
internal protocol: V1FeatureInitializable: V1Feature {
   // constrained interface of `V1FeatureInitializable`
}

protocol V1Feature {
   // not constrained V1 feature interface
}
```
that can appear in non-generic context (e.g. `v1Features: [String: V1Feature]`). I leave that door open without actually adding `V1Feature` as it requires more refactoring (e.g. if we want it to appear in `public` interface due to `DatadogCoreProtocol` being `public`).